hw/top_earlgrey/data/top_earlgrey.h.tpl

```diff
@@ -45,6 +45,17 @@
   % endif
 % endfor
 
+% for m in top["module"]:
```

Comment:
what about the memories? Do you think those base addresses are needed in software? or just link script?

Reply:
I'm not sure we need the memory addresses/ranges in software beyond the linker scripts. I can do a follow-up PR that generates the relevant parts of the linker script for us.

Reply:
It would certainly be useful for testing.

Reply:
@sriyerg if you tell me what you need, I can expand the scope of #2190 to include definitions for those memories.

Reply:
Just the base addr and lengths should suffice - `<MEM>_BASE_ADDR`, `<MEM>_SIZE`.